src/celengine/asterismrenderer.cpp

```diff
@@ -114,17 +115,18 @@ GLfloat* AsterismRenderer::prepare()
             if (chain.size() <= 1)
                 continue;
 
-            memcpy(ptr, chain[0].data(), 3 * sizeof(float));
-            ptr += 3;
-            for (unsigned i = 1; i < chain.size() - 1; i++)
+            for (unsigned i = 1; i < chain.size(); i++)
             {
-                memcpy(ptr,     chain[i].data(), 3 * sizeof(float));
-                memcpy(ptr + 3, chain[i].data(), 3 * sizeof(float));
-                ptr += 6;
+                Eigen::Vector3f prev = chain[i - 1];
+                Eigen::Vector3f cur = chain[i];
+                data.push_back({prev, cur, -0.5});
```

Comment:
it's better to use `emplace_back` here. `data.emplace_back(prev, cur, -0.5);` it works as "placement **new**".

Reply:
it seems we need to create a constructor for the struct for this purpose? otherwise it complains no matching method.

Reply:
Yes, just checked our current usage of the method. `data.emplace_back(Eigen::Vector3f(prev, cur, -0.5));` and so on

Reply:
fixed.